### PR TITLE
Migrate to operator-sdk 1.13

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

### CODE REVIEW

**Comment by pavolloffay** [comment]
Will this PR introduce new bundle format? 

With the move to SDK 1.x we should make sure the project structure is aligned with kubebuilder - I did a similar PR to oteloperator https://github.com/open-telemetry/opentelemetry-operator/pull/533. This is important to get right otherwise bundle generation does not include new fields and it is hard to add new components - API/webhooks.

**Comment by rubenvp8510** [comment]
> Will this PR introduce new bundle format?
> 
Yes
> With the move to SDK 1.x we should make sure the project structure is aligned with kubebuilder - I did a similar PR to oteloperator [open-telemetry/opentelemetry-operator#533](https://github.com/open-telemetry/opentelemetry-operator/pull/533). This is important to get right otherwise bundle generation does not include new fields and it is hard to add new components - API/webhooks.

I'll take a look, I generated this structure using the tool but I'm gonna take a look at your PR. Thanks

**Comment by rubenvp8510** [comment]
@pavolloffay  Could you please review?

@jpkrohling  If you could also take a look when you get a change it would be wonderful! :)

**Comment by rubenvp8510** [comment]
After this PR I'm planing to do other things:
 - Adapt the release process to the new bundle format
 - Clean the `Makefile`
 - Update the `README`  and `CONTRIBUTING`

File: bundle/tests/scorecard/config.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+apiVersion: scorecard.operatorframework.io/v1alpha3
```
Comment (pavolloffay) [inline bundle/tests/scorecard/config.yaml:1]: are these tests passing? I would prefer to add scorecard in a separate PR with a proper CI execution.

File: bundle/tests/scorecard/config.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+kind: Configuration
+metadata:
+  name: config
+stages:
+- parallel: true
+  tests:
+  - entrypoint:
+    - scorecard-test
+    - basic-check-spec
+    image: quay.io/operator-framework/scorecard-test:v1.12.0
```
Comment (pavolloffay) [inline bundle/tests/scorecard/config.yaml:11]: this should match the operator-sdk version

File: config/samples/jaegertracing.io_v1_jaeger.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: jaeger-sample
+spec:
+  # Add fields here
+  foo: bar
```
Comment (pavolloffay) [inline config/samples/jaegertracing.io_v1_jaeger.yaml:7]: please fix the example to be all-in-ine

File: config/samples/kafka_v1beta2_kafka.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+apiVersion: kafka.jaegertracing.io/v1beta2
```
Comment (pavolloffay) [inline config/samples/kafka_v1beta2_kafka.yaml:1]: this file should not be here and all references to `kafka.jaegertracing.io` should be removed

File: config/scorecard/bases/config.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+apiVersion: scorecard.operatorframework.io/v1alpha3
```
Comment (pavolloffay) [inline config/scorecard/bases/config.yaml:1]: As said before, it would be cleaner to add scorecard in a separate PR, hence please remove the scorecard config as well

**Review by pavolloffay — COMMENTED** [review]
A couple of notes:
* i think the kafka API should not be in the `apis` package. A, it does not align with the kubebuilder structure. B, the jaeger operator just creates this object.
* readme will have to be updated along with https://www.jaegertracing.io/docs/1.28/operator/

File: bundle/tests/scorecard/config.yaml (hunk 1 above)
Comment (rubenvp8510) [reply bundle/tests/scorecard/config.yaml:1]: I don't know this was added by the `operator-sdk` tool. The scorecard test is out of the scope of this PR.

**Comment by rubenvp8510** [comment]
> A couple of notes:
> 
>     * i think the kafka API should not be in the `apis` package. A, it does not align with the kubebuilder structure. B, the jaeger operator just creates this object.
Agree, I think the only reason for having in `apis` is for generate the `zz_generated.deepcopy.go` file
> 
>     * readme will have to be updated along with https://www.jaegertracing.io/docs/1.28/operator/

File: config/samples/jaegertracing.io_v1_jaeger.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+apiVersion: jaegertracing.io/v1
+kind: "Jaeger"
+metadata:
+  name: "my-jaeger"
+spec:
+  strategy: allInOne
+  allInOne:
+    image: jaegertracing/all-in-one:1.28
```
Comment (pavolloffay) [inline config/samples/jaegertracing.io_v1_jaeger.yaml:8]: this is not a good example. The image will get outdated soon :)

**Review by pavolloffay — COMMENTED** [review]
Could you please add as well bundle dockerfile and targets to build it https://github.com/open-telemetry/opentelemetry-operator/blob/main/bundle.Dockerfile

Note that the labels in the bundle should be updated during the `bundle` target in makefile

**Comment by rubenvp8510** [comment]
> Could you please add as well bundle dockerfile and targets to build it https://github.com/open-telemetry/opentelemetry-operator/blob/main/bundle.Dockerfile
> 
> Note that the labels in the bundle should be updated during the `bundle` target in makefile

I've already added it to the make file

It has the `bundle` and `bundle-build `target.

File: config/samples/jaegertracing.io_v1_jaeger.yaml (hunk 6 above)
Comment (rubenvp8510) [reply config/samples/jaegertracing.io_v1_jaeger.yaml]: Removed the image :)

**Review by pavolloffay — APPROVED** [review]
